File: src/com/maddyhome/idea/vim/helper/MessageHelper.java
/*
 * IdeaVim - Vim emulator for IDEs based on the IntelliJ platform
 * Copyright (C) 2003-2013 The IdeaVim authors
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 2 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */

package com.maddyhome.idea.vim.helper;

import com.intellij.CommonBundle;
import com.intellij.reference.SoftReference;
import org.jetbrains.annotations.NonNls;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;
import org.jetbrains.annotations.PropertyKey;

import java.lang.ref.Reference;
import java.util.ResourceBundle;

/**
 *
 */
public class MessageHelper {
  @Nullable private static Reference<ResourceBundle> ourBundle;

  @NonNls
  private static final String BUNDLE = "messages";

  private MessageHelper() {
  }

  @NotNull
  public static String message(@NotNull @PropertyKey(resourceBundle = BUNDLE)String key, Object... params) {
    return CommonBundle.message(getBundle(), key, params);
  }

  /*
   * This method added for jruby access
   */
  @NotNull
  public static String message(@NotNull @PropertyKey(resourceBundle = BUNDLE)String key) {
    return CommonBundle.message(getBundle(), key);
  }

  @Nullable
  protected static ResourceBundle getBundle() {
    ResourceBundle bundle = null;
    if (ourBundle != null) bundle = ourBundle.get();
    if (bundle == null) {
      bundle = ResourceBundle.getBundle(BUNDLE);
      ourBundle = new SoftReference<ResourceBundle>(bundle);
    }
    return bundle;
  }
}

File: src/main/java/com/maddyhome/idea/vim/helper/MessageHelper.kt
/*
 * Copyright 2003-2023 The IdeaVim authors
 *
 * Use of this source code is governed by an MIT-style
 * license that can be found in the LICENSE.txt file or at
 * https://opensource.org/licenses/MIT.
 */

package com.maddyhome.idea.vim.helper

import com.intellij.DynamicBundle
import org.jetbrains.annotations.NonNls
import org.jetbrains.annotations.PropertyKey

@NonNls
private const val IDEAVIM_BUNDLE = "messages.IdeaVimBundle"

internal object MessageHelper : DynamicBundle(IDEAVIM_BUNDLE) {

  private const val BUNDLE = IDEAVIM_BUNDLE

  @JvmStatic
  fun message(@PropertyKey(resourceBundle = BUNDLE) key: String, vararg params: Any) = getMessage(key, *params)

  @JvmStatic
  fun message(@PropertyKey(resourceBundle = BUNDLE) key: String) = getMessage(key)
}
